### PR TITLE
symbols: Add DB connection for gitserver client

### PR DESCRIPTION
**Pull request by eseliger**
This adds a DB connection to be used for the gitserver client in the symbols service. [A follow-up PR](https://github.com/sourcegraph/sourcegraph/pull/39504) will make better use of it. We can also use the DB for some optimizations in the future.

## Test plan

Verified locally it can still start and that metrics show up properly. Also running a main dry run.

### CODE REVIEW

File: cmd/symbols/shared/main.go [hunk 1]
```diff
@@ -93,3 +126,16 @@ func Main(setup SetupFunc) {
 	close(ready)
 	goroutine.MonitorBackgroundRoutines(context.Background(), routines...)
 }
+
+func mustInitializeFrontendDB(logger log.Logger) *sql.DB {
+	dsn := conf.GetServiceConnectionValueAndRestartOnChange(func(serviceConnections conftypes.ServiceConnections) string {
+		return serviceConnections.PostgresDSN
+	})
+
+	db, err := connections.EnsureNewFrontendDB(dsn, "symbols", &observation.TestContext)
```
Comment (ryanslade) [inline cmd/symbols/shared/main.go:135]: Is it ok to use a TestContext here?

Comment (eseliger) [reply cmd/symbols/shared/main.go:135]: It looks incorrect so I'm about to change it here for the main db.
cc @efritz you seem do be doing that for code intel DB all over the place?

File: cmd/symbols/shared/main.go [hunk 2]
```diff
@@ -67,8 +75,33 @@ func Main(setup SetupFunc) {
...
+
+		fmt.Print("Running sanity check...")
+		if err := sqlite.SanityCheck(); err != nil {
+			fmt.Println("failed ❌", err)
+			os.Exit(1)
+		}
+
+		fmt.Println("passed ✅")
+		os.Exit(0)
+	}
```
Comment (efritz) [inline cmd/symbols/shared/main.go:97]: Why did this move?

**Review by efritz — APPROVED** [review]

File: cmd/symbols/shared/main.go (hunk 2 above)
Comment (eseliger) [reply cmd/symbols/shared/main.go:97]: It must be done before the DB is initialized, as we don't have a DB where we run the check (inside docker when building the image)

**Comment by eseliger** [comment]
Looks healthy, symbols on k8s still starts up properly!

**Pull request merged**